site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java

```diff
@@ -126,4 +138,44 @@ public void postEvent(String event, String userId) {
         eventTrackingService.post(eventTrackingService.newEvent(event, userId, true/*update event*/));
     }
 
+    public Map<String, List<String>> getGroupLockingEntities (Group group) {
+        Map<String, List<String>> lockingEntitiesMap = new HashMap<String, List<String>>();
+        List<String[]> groupRealmLocks = group.getRealmLocks();
+
+        // Add one list per entity that locks groups.
+        List<String> assignmentTitles = new ArrayList<String>();
+        List<String> assessmentTitles = new ArrayList<String>();
+
+        for (String[] groupRealmLock : groupRealmLocks) {
+            String objectReference = groupRealmLock[0];
+            if (StringUtils.isNotBlank(objectReference)) {
+                if (objectReference.startsWith(GroupManagerConstants.ASN_ENTITY_PREFIX)) {
+                    String assignmentReference = objectReference.substring(objectReference.lastIndexOf("/") + 1);
```

Comment:
Use the AssignmentReferenceReckoner to load an assignment reference which you can easily get the assignment to load

Reply:
Thanks for that, implemented the change, now using the AssignmentReferenceReckoner.